ledger/catchpointwriter.go

```diff
@@ -169,6 +184,36 @@ func (cw *catchpointWriter) Abort() error {
 	return os.Remove(cw.filePath)
 }
 
+func (cw *catchpointWriter) WriteStateProofVerificationData() (crypto.Digest, error) {
+	rawData, err := StateProofVerification(cw.ctx, cw.tx)
+	if err != nil {
+		return crypto.Digest{}, err
+	}
+
+	wrappedData := catchpointStateProofVerificationData{Data: rawData}
+	encodedData := protocol.Encode(&wrappedData)
+
+	err = cw.tar.WriteHeader(&tar.Header{
+		Name: "stateProofVerificationData.msgpack",
+		Mode: 0600,
+		Size: int64(len(encodedData)),
+	})
+
+	if err != nil {
+		return crypto.Digest{}, err
+	}
+
+	_, err = cw.tar.Write(encodedData)
+	if err != nil {
+		return crypto.Digest{}, err
+	}
+
+	// TODO: This actually causes double usage of protocol.encode - above and in crypto.HashObj
+	dataHash := crypto.HashObj(wrappedData)
```

Comment:
Currently causes double usage of protocol.encode -
1. encodedData := protocol.Encode(&wrappedData)
2. dataHash := crypto.HashObj(wrappedData)
Anything we can do about it?

Reply:
yes, have a helper method on catchpointStateProofVerificationData object that takes serialized data, concats with a prefix and calls crypto.Hash()

Reply:
Went with Idan's suggestion.